### PR TITLE
Reduce template nesting

### PR DESCRIPTION
**Pull request by tempo22**
This change allow to directly access template yields to avoid having to call Field.template, making a less verbose syntax for the more complex layouts

Also added the possibility to pass @class to the f.field directly.

```
<f.field 
  @fieldName="simple_select_test" 
  @label="Simple select test"
  @control={{component "lp-select" 
    options=prospects 
    placeholder="Please select a prospect"
  }}
  as |Field Template|
>
  <Template.control as |Option|>
    <LpIcon @icon="user"/> {{Option}}
  </Template.control>
</f.field>

<f.field 
  @fieldName="simple_select_test2" 
  @label="Simple select test"
  @control="lp-text-input"
  @class="lp-c-test-class"
  as |Field Template|
>
  <Template.control />
  <Template.message>
    <LpIcon @icon="plus"/> Custom template message
  </Template.message>
</f.field>
```

### CODE REVIEW

File: addon/templates/components/light-field.hbs [hunk 1]
```diff
@@ -18,23 +31,9 @@
...
-    isDisplayingErrors=isDisplayingErrors
-    enterEditMode=(action 'enterEditMode')
-    leaveEditMode=(action 'leaveEditMode')
-    update=(action 'update')
-  )}}
+      template
+    }}
+  {{/component}}
 {{else}}
   {{component component
```
Comment (jbescoyez) [inline addon/templates/components/light-field.hbs]: @tempo22 Shouldn't it be `template` here?

Comment (tempo22) [reply addon/templates/components/light-field.hbs]: 👍

**Comment by tempo22** [comment]
> @tempo22 Tu peux me donner le code du default template pour ton deuxième exemple dans la PR? Tnx.

Voila le template du lp-field (qui est utilisé comme defaultTemplate)
```
{{#if label}}
  <label for={{controlId}} class="lp-c-label {{if required "lp-c-label--required"}}">{{label}}</label>
{{/if}}
{{#if (has-block)}}
  {{yield (hash 
    label=(component "lp-field/lp-field-element" 
      tagName="label" 
      for=controlId 
      class=(concat "lp-c-label" (if required " lp-c-label--required"))
    )
  )}}
{{/if}}

{{#if control}}
  {{#if (has-block)}}
    {{yield (hash 
      control=(component control 
        required=required
        onfocus=(action "onFocus")
        onblur=(action "onBlur")
        onchange=(action "onChange")
        value=(readonly value) 
        hasError=isDisplayingErrors
        id=controlId
      )
    )}} 
  {{else}}
    {{component control 
      required=required
      onfocus=(action "onFocus")
      onblur=(action "onBlur")
      onchange=(action "onChange")
      value=(readonly value)
      hasError=isDisplayingErrors
      id=controlId
    }}
  {{/if}}
{{else}}
  {{#if (has-block)}}
    {{yield (hash 
      control=(component "lp-field/lp-field-element" tagName="")
    )}}
  {{else}}
    {{component "lp-text-input" 
      autofocus=autofocus
      required=required
      placeholder=placeholder
      onfocus=(action enterEditMode)
      onblur=(action leaveEditMode)
      onchange=(action "onChange")
      value=(readonly value) 
      hasError=isDisplayingErrors
      id=controlId
    }}
  {{/if}}
{{/if}}

{{#if (has-block)}}
  {{yield (hash message=(component "lp-field/lp-field-element" tagName="p" class="lp-c-field__message"))}} 
{{else if message}}
  {{#component "lp-field/lp-field-element" tagName="p" class="lp-c-field__message"}}
    {{message}}
  {{/component}}
{{/if}}

{{#if isDisplayingErrors}}
  {{#component "lp-field/lp-field-element" tagName="p" class="lp-c-field__message lp-c-field__message--error"}}
    {{errors}}
  {{/component}}
{{/if}}
```

**Comment by jbescoyez** [comment]
Ok. I cannot merge immediately. I first have to test with simpler templates. Will do that tomorrow.

**Comment by tempo22** [comment]
> Ok. I cannot merge immediately. I first have to test with simpler templates. Will do that tomorrow.

👍 Also this is a breaking change for most field that already have the complex syntax implemented.

**Comment by tempo22** [comment]
Think of it, if we leave the `component=(component component` in the yielded hash, it won't break existing templates.

**Pull request merged**